src/Nethermind/Nethermind.Evm/Precompiles/Stateful/BeaconBlockRootPrecompile.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+using Nethermind.State;
+
+namespace Nethermind.Evm.Precompiles.Stateful;
+public class BeaconBlockRootPrecompile : IPrecompile
+{
+    public static Address Address { get; } = Address.FromNumber(0x0B);
+    public static UInt256 HISTORICAL_ROOTS_LENGTH = 98304;
+
+    public static IPrecompile Instance => new BeaconBlockRootPrecompile();
+
+    private byte[] SloadFromStorage(IWorldState state, in UInt256 index)
+    {
+        StorageCell storageCell = new(Address, index);
+        return state.Get(storageCell);
+    }
+    public static void SetupBeaconBlockRootPrecompileState(IWorldState stateProvider, Keccak parentBeaconBlockRoot, UInt256 timestamp)
```

Comment:
Have you still planned to refactor it from being static?

Reply:
yes, I am trying ways to do it and see which seems better